pkg/cloud/services/securitygroup/securitygroups.go

```diff
@@ -743,15 +817,15 @@ func ingressRulesFromSDKType(v *ec2.IpPermission) (res infrav1.IngressRules) {
 	}
 
 	if len(v.IpRanges) > 0 {
-		r1 := ir
 		for _, ec2range := range v.IpRanges {
+			r1 := ir
```

Comment:
I'm not entirely sure this is an accurate fix. 🤔 

It's supposed to add to the CidrBlocks of r1 than at the end add r1 to the results. This keeps adding multiple r1s to the result. It's not the same as before. 🤔